tests/basic/tracing.rs

```diff
@@ -0,0 +1,127 @@
+use proptest::proptest;
+use proptest::test_runner::Config;
+use shuttle::future::spawn;
+use shuttle::sync::{Arc, Mutex};
+use shuttle::{check_random, thread};
+use std::time::Duration;
+use test_log::test;
+use tracing::{instrument::Instrument, warn, warn_span};
+
+// NOTE:
+// All of the tests testing tracing will either have to have weak assertions,
+// or be #[ignore]d. The reason for this is that they are not thread safe
+// (since everything tracing is managed globally), and there is no way to
+// ensure that the tests are run single-threaded.
```

Comment:
There is this crate: https://docs.rs/serial_test/latest/serial_test/index.html

See if it's worth using here?

Reply:
Will do, thanks!

Reply:
It seems to me that it won't solve the issue, unless all other tests are marked with `parallel`, which isn't worth it. This is because marking a test with `serial` only guarantees that it won't be ran at the same time as other `serial` or `parallel` tests, but it can still run at the same time as tests which are not marked.